src/styles/calendar.scss

```
@@ -0,0 +1,55 @@
+@use './variables' as *;
+
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+  padding: $calendar-padding;
+  width: $calendar-flex-width;
+  min-height: $calendar-flex-height;
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: $days-color;
+    width: $calendar-size;
+    height: $calendar-size;
+    border: $calendar-border solid black;
+    transition: 0.5s ease;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+        font-family: $calendar-font;
+        font-size: $calendar-font-size;
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: $hover-color;
+      transform: translateY(-$calendar-animation-size);
+    }
+  }
+
+  @each $day in $week-days {
+    &--start-day#{$day} {
+      .calendar__day:nth-child(1) {
+        margin-left: $counter * $base-spacing;
+      }
+    }
+
+    $counter: $counter + 1;
+  }
```

instead of counter you can add values next to all week-days
```suggestion
  @each $day, $day-number in $week-days {
    &--start-day-#{$day} {
      .calendar__day:nth-child(1) {
        margin-left: $day-number * $base-spacing;
      }
    }
  }
```

---

Yes, right.. I thought it would be better to add a counter to reduce code writing, since adding a counter variable would be more universal.